### PR TITLE
CSHARP3725: Fix TestThatItRunsConventionsInTheProperOrder test

### PR DESCRIPTION
**Pull request by BorisDog**
[EG](https://evergreen.mongodb.com/version/60ef764357e85a201911bb00)

### CODE REVIEW

File: tests/MongoDB.Bson.Tests/Serialization/Conventions/ConventionRunnerTests.cs [hunk 1]
```diff
@@ -140,17 +142,17 @@ public void Apply(BsonClassMap classMap)
             {
                 IsRun = true;
                 RunCount++;
-                RunTicks = _stopwatch.ElapsedTicks;
+                RunTicks = _orderIndexProvider();
```
Comment (JamesKovacs) [inline tests/MongoDB.Bson.Tests/Serialization/Conventions/ConventionRunnerTests.cs:145]: This isn't `RunTicks` anymore. It is `RunOrder`.

File: tests/MongoDB.Bson.Tests/Serialization/Conventions/ConventionRunnerTests.cs [hunk 2]
```diff
@@ -48,9 +50,9 @@ public ConventionRunnerTests()
                 cm.MapMember(t => t.Prop2);
             });
 
-            stopwatch.Start();
             _subject.Apply(classMap);
-            stopwatch.Stop();
+
+            int GetIndex() => Interlocked.Increment(ref orderIndex);
```
Comment (JamesKovacs) [inline tests/MongoDB.Bson.Tests/Serialization/Conventions/ConventionRunnerTests.cs:55]: `GetRunOrderIndex` is more descriptive.

**Review by JamesKovacs — CHANGES_REQUESTED** [review]
Minor naming changes requested for clarity, but the technique is sound.

File: tests/MongoDB.Bson.Tests/Serialization/Conventions/ConventionRunnerTests.cs (hunk 2 above)
Comment (BorisDog) [reply tests/MongoDB.Bson.Tests/Serialization/Conventions/ConventionRunnerTests.cs]: Done

File: tests/MongoDB.Bson.Tests/Serialization/Conventions/ConventionRunnerTests.cs (hunk 1 above)
Comment (BorisDog) [reply tests/MongoDB.Bson.Tests/Serialization/Conventions/ConventionRunnerTests.cs]: Done

**Review by JamesKovacs — APPROVED** [review]
LGTM

**Comment by BorisDog** [comment]
[Merged](https://github.com/mongodb/mongo-csharp-driver/commit/b1831c2d0b0ad4807daf9fdeaabd077b73ed3f79)

**Pull request closed without merging**